api/src/main/java/org/openmrs/module/clientregistry/FhirConfig.java

```diff
@@ -0,0 +1,32 @@
+package org.openmrs.module.clientregistry;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.client.interceptor.BasicAuthInterceptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class FhirConfig {
+	
+	@Autowired
+	private ClientRegistryConfig config;
+	
+	@Autowired
+	@Qualifier("fhirR4")
+	private FhirContext fhirContext;
+	
+	@Bean
+	public IGenericClient getFhirClient() throws Exception {
+		IGenericClient fhirClient = fhirContext.newRestfulGenericClient(config.getClientRegistryServerUrl());
+		if (!config.getClientRegistryUserName().isEmpty()) {
+			BasicAuthInterceptor authInterceptor = new BasicAuthInterceptor(config.getClientRegistryUserName(),
```

Comment:
should we be using basic auth? im ok keeping this temporarily but i think its worth considering token based authentication if possible

Reply:
We should create an issue for adding certificate-based authentication like in the labonfhir module: https://github.com/openmrs/openmrs-module-labonfhir/blob/ff3b970079ee39b5a69554fa426fbc2a3ba8ecc4/api/src/main/java/org/openmrs/module/labonfhir/FhirConfig.java#L35